src/main/java/org/spongepowered/common/util/FireworkUtil.java

```diff
@@ -32,37 +32,47 @@
 import net.minecraft.world.item.component.Fireworks;
 import org.spongepowered.api.item.FireworkEffect;
 import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.common.accessor.world.entity.EntityAccessor;
 import org.spongepowered.common.accessor.world.entity.projectile.FireworkRocketEntityAccessor;
 import org.spongepowered.common.item.SpongeItemStack;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.function.Function;
 
 public final class FireworkUtil {
+    public static boolean setFireworkEffects(final FireworkRocketEntity firework, final List<? extends FireworkEffect> effects) {
+        return updateFireworkRocketItem(firework, item -> setFireworkEffects(item, effects));
```

Comment:
Keep the type qualifiers, there are few others.

```suggestion
        return FireworkUtil.updateFireworkRocketItem(firework, item -> FireworkUtil.setFireworkEffects(item, effects));
```